BZ2096868: Add steps to restore to a previous cluster state (#50076)
openshift/openshift-docs: modules/dr-restoring-cluster-state.adoc

If the status is `Pending`, or the output lists more than one running etcd pod, wait a few minutes and check again.

.. Repeat this step for each lost control plane host that is not the recovery host.
+
[NOTE]
====

> Perform the steps from 12 to 15 only if you are using `OVNKubernetes` Container Network Interface (CNI) plugin.

Author: Addressed

====
+

> . Remove the northbound database (nbdb) and southbound database (sbdb). Access the recovery host and the remaining control plane nodes by using Secure Shell (SSH) and run the following command:

Reviewer: ```suggestion
. Restart the OVN Kubernetes pods on all the hosts.

.. Remove the northbound database (nbdb) and southbound database (sbdb). Access the recovery host and the remaining control plane nodes by using Secure Shell (SSH) and run the following command:
```

Author: Addressed

If the status is `Pending`, or the output lists more than one running etcd pod, wait a few minutes and check again.

.. Repeat this step for each lost control plane host that is not the recovery host.
+
[NOTE]
====
Perform the following step only if you are using `OVNKubernetes` Container Network Interface (CNI) plugin.
====
+
. Restart the Open Virtual Network (OVN) Kubernetes pods on all the hosts.

.. Remove the northbound database (nbdb) and southbound database (sbdb). Access the recovery host and the remaining control plane nodes by using Secure Shell (SSH) and run the following command:
+
[source,terminal]
----
$ sudo rm -f /var/lib/ovn/etc/*.db
----

.. Delete all master pods by running the following command:
+
[source,terminal]
----
$ oc delete pods -l app=ovnkube-master -n openshift-ovn-kubernetes
----

> .. Ensure that all pods are running by running the following command:

Reviewer: ```suggestion
.. Ensure that all the master pods are deployed again and are in Running state by running the following command:
```

Author: Addressed

> .. Ensure that all the `ovnkube-node` pods are deployed again and are in `Running` state by running the following command:
> +
> [source,terminal]
> ----
> $ oc get  pods -n openshift-ovn-kubernetes | grep ovnkube-node

Reviewer: ```suggestion
.. Ensure that all the master pods are deployed again and are in `Running` state by running the following command:
+
[source,terminal]
----
$ oc get  pods -l app=ovnkube-master -n openshift-ovn-kubernetes
```

----
+
.Example output
[source,terminal]
----
NAME                   READY   STATUS    RESTARTS   AGE
ovnkube-master-nb24h   4/4     Running   0          48s
ovnkube-master-rm8kw   4/4     Running   0          47s
ovnkube-master-zbqnh   4/4     Running   0          56s
----

> .. Delete all ovn-kubenode pods by running the following command:

Reviewer: ```suggestion
.. Delete all ovnkube-node pods by running the following command:
```

Author: Addressed

+
[source,terminal]
----
$ oc get pods -n openshift-ovn-kubernetes -o name | grep ovnkube-node | while read p ; do oc delete $p -n openshift-ovn-kubernetes ; done
----

> 

Reviewer: ```suggestion
.. Ensure that all the ovnkube-node pods are deployed again and are in Running state by running the following command:
+
[source,terminal]
----
$ oc get  pods -l app=ovnkube-node  -n openshift-ovn-kubernetes
----
```

If the status is `Pending`, or the output lists more than one running etcd pod, wait a few minutes and check again.

.. Repeat this step for each lost control plane host that is not the recovery host.
+
[NOTE]
====

> Perform the following step only if you are using `OVNKubernetes` Container Network Interface (CNI) plugin.

Author: acked

====
+
. Restart the Open Virtual Network (OVN) Kubernetes pods on all the hosts.

.. Remove the northbound database (nbdb) and southbound database (sbdb). Access the recovery host and the remaining control plane nodes by using Secure Shell (SSH) and run the following command:
+
[source,terminal]
----
$ sudo rm -f /var/lib/ovn/etc/*.db
----

> .. Delete all master pods by running the following command:

Reviewer 2: Is there any way to [avoid using the term "master"](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#using-conscious-language) here?

Reviewer: Sure. We can use "control-plane" instead of "master". It is also followed here: https://docs.openshift.com/container-platform/4.11/post_installation_configuration/network-configuration.html#nw-ovn-ipsec-verification_post-install-network-configuration

Author: addressed

> .. Delete all master pods by running the following command:

Reviewer: ```suggestion
.. Delete all OVN-Kubernetes control plane pods by running the following command:
```

Author: acked

+
[source,terminal]
----
$ oc delete pods -l app=ovnkube-master -n openshift-ovn-kubernetes
----

> .. Ensure that all the master pods are deployed again and are in a `Running` state by running the following command:

Reviewer 2: See above question

Author: addressed

> .. Ensure that all the master pods are deployed again and are in a `Running` state by running the following command:

Reviewer: ```suggestion
.. Ensure that all the OVN-Kubernetes control plane pods are deployed again and are in a `Running` state by running the following command:
```

Author: acked

+
[source,terminal]
----
$ oc get pods -l app=ovnkube-master -n openshift-ovn-kubernetes
----
+
.Example output
[source,terminal]
----
NAME                   READY   STATUS    RESTARTS   AGE
ovnkube-master-nb24h   4/4     Running   0          48s
ovnkube-master-rm8kw   4/4     Running   0          47s
ovnkube-master-zbqnh   4/4     Running   0          56s
----

> .. Delete all `ovn-kubenode` pods by running the following command:

Author: acked

.. Repeat this step for each lost control plane host that is not the recovery host.

> . Remove the northbound database (nbdb) and southbound database (sbdb). Access the master nodes and recovery host by using SSH and run the following command:

Reviewer: ```suggestion
. Remove the northbound database (nbdb) and southbound database (sbdb). Access the recovery host and the remaining master nodes by using SSH and run the following command:
```

Author: Addressed - Changed the terminology for master nodes ( control plane nodes ) as per style guide rules.

+
[source,terminal]
----
$ for p in master-0 master-1 master-2;do ssh core@$p sudo rm -f /var/lib/ovn/etc/*.db;done

> ----

Author: addressed

. Delete all master pods by running the following command:
+
[source,terminal]
----

> $ oc delete pods -l app=ovnkube-master

Author: acked

----

. Ensure that all pods are running by running the following command:
+
[source,terminal]
----

> $ oc get  pods -l app=ovnkube-master

Reviewer: ```suggestion
$ oc get  pods -l app=ovnkube-master -n openshift-ovn-kubernetes
```

Author: Addressed

----
+
.Example output
[source,terminal]
----
NAME                   READY   STATUS    RESTARTS   AGE

> ovnkube-master-nb24h   4/4     Running   0          48s master
> ovnkube-master-rm8kw   4/4     Running   0          47s follower
> ovnkube-master-zbqnh   4/4     Running   0          56s follower

Reviewer: I don't think we get which `ovnkube-master` pod is the master and which ones are follower from the above `oc get pods` command
```suggestion
ovnkube-master-nb24h   4/4     Running   0          48s
ovnkube-master-rm8kw   4/4     Running   0          47s
ovnkube-master-zbqnh   4/4     Running   0          56s
```

Author: Acked

----

. Delete all ovn-kubenode pods by running the following command:
+
[source,terminal]
----

> $ oc get pods -o name | grep ovnkube-node | while read p ; do oc delete $p ; done

Author: acked
